src/Arcus.Scripting.Tests.Unit/Arcus.Scripting.ApiManagement.tests.ps1

```diff
@@ -143,14 +150,55 @@ InModuleScope Arcus.Scripting.ApiManagement {
+                Backup-AzApiManagementService -ResourceGroupName $resourceGroup -StorageAccountResourceGroup $storageAccountResourceGroup -StorageAccountName $expectedStorageAccountName -ServiceName $serviceName -ContainerName $targetContainerName -AccessType $accessType -PassThru -DefaultProfile $defaultProfile
+
+                # Assert
+                Assert-VerifiableMock
+                Assert-MockCalled Get-AzStorageAccountKey -Times 1
+                Assert-MockCalled New-AzStorageContext -Times 1
+                Assert-MockCalled Backup-AzApiManagement -Times 1
+            }
+            It "Backs up API management with User Assigned Managed Identity" {
+                # Arrange
+                $resourceGroup = "shopping"
+                $storageAccountResourceGroup = "stock"
+                $expectedStorageAccountName = "shopping-storage"
+                $serviceName = "shopping-API-management"
+                $targetContainerName = "backup-storage"
+                $storageKeyValue = "my-storage-key"
+                $storageKey = New-Object -TypeName Microsoft.Azure.Management.Storage.Models.StorageAccountKey -ArgumentList @($null, $storageKeyValue, $null)
+                $accessType = 'UserAssignedManagedIdentity'
+                $identityClientId = 'someidentityid'
+
+                Mock Get-AzStorageAccountKey {
+                    $ResourceGroupName | Should -Be $storageAccountResourceGroup
+                    $StorageAccountName | Should -Be $expectedStorageAccountName
+                    return $storageKey }
+                Mock New-AzStorageContext { 
+                    $StorageAccountName | Should -Be $StorageAccountName
+                    $StorageAccountKey | Should -Be $storageKeyValue
+                    return $expectedStorageContext }
+                Mock Backup-AzApiManagement {
+                    $ResourceGroupName | Should -Be $resourceGroup
+                    $Name | Should -Be $serviceName
+                    $AccessType | Should -Be $accessType
+                    $IdentityClientId | Should -Be $identityClientId
+                    $StorageContext | Should -be $expectedStorageContext
+                    $TargetContainerName | Should -Be $targetContainerName
+                    $TargetBlobName | Should -BeNullOrEmpty 
+                    $PassThru | Should -Be $false
+                    $DefaultProfile | Should -Be $null }
+
+                # Act
+                Backup-AzApiManagementService -ResourceGroupName $resourceGroup -StorageAccountResourceGroup $storageAccountResourceGroup -StorageAccountName $expectedStorageAccountName -ServiceName $serviceName -ContainerName $targetContainerName -AccessType $accessType -IdentityClientId $identityClientId
```

Comment:
We can't integration test this yet bc it is too expensive, right? Shame, then this is indeed best we can do. Thx!

Reply:
Indeed, bit too expensive